### ISSUE TITLE
Reconcile classic and compat versions

### ISSUE THREAD

**Issue by fabiocicerchia** [fabiocicerchia/nginx-lua#65]
There are currently 2 versions, classic (using the lua system package) and the compat (building from source).
The classic version is not really working correctly with luarocks and modules, so it's time to restructure the 2 versions to be only 1, which is fully working without having multiple versions.

In terms of libraries the classic version is using Lua 5.4, while compat is using LuaJIT 2.1 (provided by OpenResty):

```
Files nginx-classic/usr/include/lua5.4/lauxlib.h and nginx-compat/usr/local/include/luajit-2.1/lauxlib.h differ
Files nginx-classic/usr/include/lua5.4/luaconf.h and nginx-compat/usr/local/include/luajit-2.1/luaconf.h differ
Files nginx-classic/usr/include/lua5.4/lua.h and nginx-compat/usr/local/include/luajit-2.1/lua.h differ
Files nginx-classic/usr/include/lua5.4/lua.hpp and nginx-compat/usr/local/include/luajit-2.1/lua.hpp differ
Only in nginx-compat/usr/local/include/luajit-2.1: luajit.h
Files nginx-classic/usr/include/lua5.4/lualib.h and nginx-compat/usr/local/include/luajit-2.1/lualib.h differ
Files nginx-classic/usr/lib/lua5.4/liblua.a and nginx-compat/usr/local/lib/liblua.a differ
```

^ extracted from `fabiocicerchia/nginx-lua:1.25.1-alpine3.18.2` vs `fabiocicerchia/nginx-lua:1.25.1-alpine3.18.2-compat`

nginx-classic is using latest lua version [5.4.6](https://github.com/lua/lua/blob/v5.4.6/lauxlib.h), nginx-compat is using open-resty luajit version [v2.1-20230410](https://github.com/openresty/luajit2/blob/v2.1-20230410/src/lauxlib.h). The OpenResty library is kind of mandatory as I'm not going to rewrite the whole underlying layer to have LuaJIT to support 5.4 (it'd rather be a separate project).

Nginx uses [`lua-nginx-module`](https://github.com/openresty/lua-nginx-module) to enable Lua support, which internally uses the [OpenResty's LuaJIT](https://github.com/openresty/luajit2) library instead of the [official LuaJIT](https://github.com/LuaJIT/LuaJIT) library.

---

**Proposal:** LuaJIT, regardless of which implementations (ie OpenResty and official one), is fixed on Lua 5.1, the whole system should be based on 5.1 to avoid confusion with versions. most likely the system package 5.4 should be downgraded to 5.1 or even dropped in favour of LuaJIT as executable.

**Comment by fabiocicerchia** [comment]
need to check if it would be possible to drop support for `/usr/local/share/lua/5.1` as a symlink to `/usr/local/share/lua/5.4`.

**Comment by fabiocicerchia** [comment]
LuaJIT (which is used by nginx) is fixed on 5.1, the whole system should be based on 5.1 to avoid confusion with versions. most likely the system package 5.4 should be downgraded to 5.1 or even dropped in favour of LuaJIT as executable.

**Comment by fabiocicerchia** [comment]
LuaJIT (which is used by nginx) is fixed on 5.1, the whole system should be based on 5.1 to avoid confusion with versions. most likely the system package 5.4 should be downgraded to 5.1 or even dropped in favour of LuaJIT as executable.
